libraries/Microsoft.Bot.Connector/Authentication/BuiltinBotFrameworkAuthentication.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Microsoft.Rest;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    internal abstract class BuiltinBotFrameworkAuthentication : BotFrameworkAuthentication
+    {
+        private readonly string _toChannelFromBotOAuthScope;
+        private readonly string _loginEndpoint;
+        private readonly string _callerId;
+        private readonly string _channelService;
+        private readonly ServiceClientCredentialsFactory _credentialFactory;
+        private readonly AuthenticationConfiguration _authConfiguration;
+        private readonly HttpClient _httpClient;
+        private readonly ILogger _logger;
```

Comment:
Will this _logger be used anywhere?

Reply:
I think so. As I say this is just phase one. I think it might be nice to log the _path_ of the auth validation for example, whether it's emulator, skills, channel service etc.

Reply:
okay, thanks @johnataylor